protocolapi/block/api.go

```diff
@@ -0,0 +1,160 @@
+package block
+
+import (
+	"context"
+	"time"
+
+	"gx/ipfs/QmNf3wujpV2Y7Lnj2hy2UrmuX8bhMDStRHbnSLh7Ypf36h/go-hamt-ipld"
+	"gx/ipfs/QmRu7tiRnFk9mMPpVECQTBQJqXtmG132jJxA1w9A7TtpBz/go-ipfs-blockstore"
+
+	"github.com/filecoin-project/go-filecoin/actor/builtin"
+	"github.com/filecoin-project/go-filecoin/address"
+	"github.com/filecoin-project/go-filecoin/chain"
+	"github.com/filecoin-project/go-filecoin/consensus"
+	"github.com/filecoin-project/go-filecoin/core"
+	"github.com/filecoin-project/go-filecoin/mining"
+	"github.com/filecoin-project/go-filecoin/porcelain"
+	"github.com/filecoin-project/go-filecoin/sampling"
+	"github.com/filecoin-project/go-filecoin/state"
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+// API provides an interface to protocols.
+type API struct {
+	apiDeps
+}
+
+type apiDeps struct {
+	addNewBlockFunc   func(context.Context, *types.Block) (err error)
```

Comment:
ugh, half these deps are for the default worker. it would be really nice to introduce a deps struct to the worker, then we could pass in its deps here in that struct and be able to see the actual deps of this dude. would you mind filing an issue that we can use to track protocol cleanups that we want to do after we kill the node? this one should go in that list, and there is one below.

Reply:
> would you mind filing an issue that we can use to track protocol cleanups that we want to do after we kill the node? this one should go in that list, and there is one below.

Ok: #2276